compiler/modules/front.syntax/parseTypeDesc.bal

```diff
@@ -277,27 +277,30 @@ function parseTypeParam(Tokenizer tok) returns TypeDesc|err:Syntax {
 }
 
 // current token should be "("
-function parseFunctionTypeDesc(Tokenizer tok, string[]? paramNames = ()) returns FunctionTypeDesc|err:Syntax {
+function parseFunctionTypeDesc(Tokenizer tok, FunctionParam[]? params = ()) returns FunctionTypeDesc|err:Syntax {
     // skip "function"
     Position startPos = tok.currentStartPos();
     check tok.expect("(");
-    TypeDesc[] args = [];
+    FunctionTypeParam[] typeParams = params ?: [];
```

Comment:
This is now very nice!

But I think you were right to use `params` for this variable.  So we need another name for the `params` parameter. Perhaps we could call it `namedParams`.

Reply:
fixed